### PR TITLE
crimson/os/seastore: throttle concurrent transactions in seastore

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Fixes: https://tracker.ceph.com/issues/57215

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/common/options/crimson.yaml.in [hunk 1]
```diff
@@ -87,3 +87,8 @@ options:
   level: advanced
   desc: split extent if ratio of total extent size to write size exceeds this value
   default: 1.25
+- name: seastore_max_concurrent_transactions
+  type: uint
+  level: advanced
+  desc: maximum concurrent transactions that seastore allows
+  default: 8
```
[the remark this answers is not in the document]
Comment (xxhdx1985126) [reply src/common/options/crimson.yaml.in:94]: According to our tests, 8 is enough to saturate a single core seastore instance.

File: src/crimson/os/seastore/seastore.cc [hunk 2]
```diff
@@ -161,6 +163,19 @@ void SeaStore::register_metrics()
       }
     );
   }
+
+  metrics.add_group(
+    "seastore",
+    {
+      sm::make_gauge(
+	"transactions",
```
[the remark this answers is not in the document]
Comment (xxhdx1985126) [reply src/crimson/os/seastore/seastore.cc]: done.

File: src/common/options/crimson.yaml.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (cyx1231st) [reply src/common/options/crimson.yaml.in:94]: I'm wondering if there is an easy way to diagnose whether this configuration is too low/high.

Probably worth adding another metric "pending_transactions". So we can be awared that this value is too low, if the CPU is not saturized and there are pending transactions.

[the remark this answers is not in the document]
Comment (xxhdx1985126) [reply src/common/options/crimson.yaml.in:94]: done.

**Review by cyx1231st — APPROVED** [review]
LGTM

**Comment by cyx1231st** [comment]
jenkins test api

**Comment by cyx1231st** [comment]
jenkins test api

**Comment by cyx1231st** [comment]
jenkins test api

**Comment by cyx1231st** [comment]
jenkins test api

**Pull request merged**